File: JavaBasics/Exams/_007_ JavaScriptBasicsExam_28_July_2014/src/_01_DoubleRakiyaNumbers.java
import java.util.Scanner;

public class _01_DoubleRakiyaNumbers {

	public static void main(String[] args) {
		@SuppressWarnings("resource")
		Scanner input = new Scanner(System.in);
		int start = Integer.parseInt(input.nextLine());
		int end = Integer.parseInt(input.nextLine());
		System.out.println("<ul>");
		for (int i = start; i <= end; i++) {
			if (IsDoubleRakiyaNumbers(new Integer(i).toString())) {
				System.out.printf("<li><span class='rakiya'>%s</span><a href=\"view.php?id=%s\">View</a></li>\n", i,i);
			} else {
				System.out.printf("<li><span class='num'>%s</span></li>\n", i);
			}
		}
		System.out.println("</ul>");
	}

	private static boolean IsDoubleRakiyaNumbers(String number) {
		char[] nums = number.toCharArray();
		if (nums.length < 4) {
			return false;
		}
		
		for (int i = 0; i <= nums.length - 4; i++) {
			String curr1 = "" + nums[i] + nums[i + 1];
			for (int k = i + 2; k <= nums.length - 2; k++) {
				String curr2 = "" + nums[k] + nums[k + 1];
				if (curr1.equals(curr2)) {
					return true;
				}
			}
		}
		return false;
	}
}

File: 03-JavaScript-Basics/Homework/03_JS_Loops/20_DoubleRakiyaNumbers/01_DoubleRakiyaNumbers.js
function printDoubleRakiaNumbers (inputStr){
    var start = parseInt(inputStr[0]);
    var end = parseInt(inputStr[1]);
    console.log("<ul>");
    for(var i = start ; i <= end; i+=1){
        var rowStr = "<li><span class=\'num\'>" + i + "</span></li>";
        if(isDoubleRakiaNum(i.toString())){
            var rowStr = "<li><span class=\'rakiya\'>" + i + "</span>" +
                "<a href=\"view.php?id=" + i + ">View</a></li>";
        }
        console.log(rowStr);
    }
    console.log("</ul>");
    function isDoubleRakiaNum (numberStr) {
        if (numberStr.length < 4){
            return false;
        }
        for(var i = 0; i <= numberStr.length - 4; i+=1){
            var count = 0;
            for(var k = i + 2; k <= numberStr.length - 2; k+=1){
                if((numberStr[i] + numberStr[i+1]) == (numberStr[k] + numberStr[k+1])){
                    return true;
                }
            }
        }
        return false;
    }
}

